client/core/trade.go

```diff
@@ -1267,7 +1329,89 @@ func (t *trackedTrade) finalizeRedeemAction(match *matchTracker, coinID []byte)
 	return errs.ifAny()
 }
 
+// beginFindRedemption begins a search for the redemption of taker's contract
+// and waits for the search results on a goroutine.
+//
+// This method modifies trackedTrade fields and MUST be called with the
+// trackedTrade mutex lock held for writes.
+func (t *trackedTrade) beginFindRedemption(match *matchTracker) {
+	swapCoinID := dex.Bytes(match.MetaData.Proof.TakerSwap)
+	resultChan, err := t.wallets.fromWallet.FindRedemption(swapCoinID)
+	if err != nil {
+		fromID := t.wallets.fromAsset.ID
+		log.Errorf("failed to start redemption search for match %v, %s contract %s: %v",
+			match.id, unbip(fromID), coinIDString(fromID, swapCoinID), err)
+		return
+	}
+	// Redemption search started, start goroutine to monitor result channel.
+	t.awaitingRedemption[swapCoinID.String()] = match.id
+	go t.waitForRedemption(resultChan)
+}
+
+// waitForRedemption waits on the passed `redemptionChan` for FindRedemption
+// results. Should be run as a goroutine.
+func (t *trackedTrade) waitForRedemption(redemptionChan chan *asset.FindRedemptionResult) {
+	frr := <-redemptionChan
+	if frr == nil {
+		return // channel closed without data
+	}
+
+	fromAsset := t.wallets.fromAsset
+	var match *matchTracker
+
+	t.mtx.Lock()
+	defer t.mtx.Unlock()
+
+	orderID, found := t.awaitingRedemption[frr.ContractCoinID.String()]
+	if found {
+		delete(t.awaitingRedemption, frr.ContractCoinID.String())
+		match, found = t.matches[orderID]
+	}
+	if !found {
+		log.Warnf("received find redemption result for unknown contract coin (%s: %s) on order %s",
+			fromAsset.Symbol, coinIDString(fromAsset.ID, frr.ContractCoinID), t.ID())
+		return
+	}
+
+	if frr.Err != nil {
+		log.Errorf("error finding redemption of taker's %s contract %s for order %s, match %s: %v",
```

Comment:
Would expect this would trigger a refund attempt, unless refunds were always tried before find redeem, but that does not seem to be the case with the current switch in `tick`.